lib/client.js

```diff
@@ -360,7 +360,7 @@ function retry(fn, ctx) {
   function attempt(i) {
     return fn(ctx)
       .catch((err) => {
-        if (maxAttempts > 0) {
+        if (i < maxAttempts && isCriticalError(err) && ctx.retryDelay && ctx.retryDelay > 0) {
```

Comment:
I wonder whether it's worth leaving a comment explaining the rationale here?

Reply:
yeah agree with this

Reply:
Re: critical errors. In this instance isn't it clear from the code that that's the case?

Reply:
@cjewell47 there are a lot of other conditions chained in the if statement, but none of them matter unless the error is critical (which I think is the main fix/change in this PR). The comment would just highlight that importance, so that someone new to the code takes note 😃

Reply:
Thanks guys, we have added some jsdoc to make this a bit clearer